docs/adr/0004-framework.md

```diff
@@ -0,0 +1,68 @@
+# Use React Framework
+
+* Status: proposed
+* Deciders: Monitor team
+* Date: 2022-05-03
+* Technical Story: [MNTOR-1588](https://mozilla-hub.atlassian.net/browse/MNTOR-1588)
+
+## Context and Problem Statement
+
+The Monitor team is tasked with launching a premium product offering as soon as possible. The current architecture of Monitor is not suited for synchronous feature development and lacks the complex client-side state management premium will require. 
+
+
+## Decision Drivers <!-- optional -->
+
+* The expansion in Monitor premium requires a framework to manage increased complexity and client-side state management
+* Promote component sharing between S&P web-based products, Relay & Monitor
+* Allow developers to work on specific components and features without affecting the rest of the codebase (fewer merge conflicts, makes it easier break down tickets)
+* Unidirectional (one-way) data flow makes it easier to understand the flow of data in the application, which may be necessary when handling multiple states (especially for the data removal sequence)
+* Monitor Front-end engineers have extensive React experience
```

Comment:
Extra context: @codemist @Vinnl and myself have valuable experience from when we decoupled the frontend on Relay from Django to React.